src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -223,6 +223,10 @@ private static bool IsTupleExpression(BoundKind kind)
             {
                 var field = fields[i];
 
+                // Use default field rather than implicitly named fields since
+                // fields from inferred names are not usable in C# 7.0.
+                field = field.CorrespondingTupleField ?? field;
```

Comment:
When would `CorrespondingTupleField` ever be non-null here?

Also can you give an example of when the use site diagnostic check below would ever get hit with this change?